sklearn/linear_model/tests/test_quantile.py

```diff
@@ -143,22 +147,23 @@ def test_asymmetric_error(quantile):
     )
     intercept = 1.23
     coef = np.array([0.5, -2])
+    assert np.max(X @ coef + intercept) > 0, "take care that X @ coef + intercept > 0"
     # For an exponential distribution with rate lambda, e.g. exp(-lambda * x),
     # the quantile at level q is:
     #   quantile(q) = - log(1 - q) / lambda
-    #   scale = 1/lambda = -quantile(q) / log(1-q)
+    #   scale = 1/lambda = -quantile(q) / log(1 - q)
     y = rng.exponential(
         scale=-(X @ coef + intercept) / np.log(1 - quantile), size=n_samples
     )
     model = QuantileRegressor(
         quantile=quantile,
         alpha=0,
-        solver="interior-point",
-        solver_options={"tol": 1e-5},
+        solver="highs",
     ).fit(X, y)
+
     assert model.intercept_ == approx(intercept, rel=0.2)
     assert_allclose(model.coef_, coef, rtol=0.6)
-    assert_allclose(np.mean(model.predict(X) > y), quantile)
+    assert_allclose(np.mean(model.predict(X) > y), quantile, rtol=6e-3)
```

Comment:
For the other assertions on the coefs and intercept, I am not sure. It seems that with `n_samples=1000`, results can vary significantly without impacting the training loss too much (ill-conditioning maybe?). For 10x larger number of samples, the test is significantly more stable but then it is slow again, even with `solver="highs"`.

@lorentzenchr Maybe we can remove the assertions on the coefs and intercept to make the test more independent of the seed and therefore easier to maintain. I am not sure we can expect identifyability on finite training sets far from the asymptotic regime.

Reply:
I'm ok with `atol ~< 1e-2`. The tests/assertions on `coef` and `intercept`, however, are veeeery essential, IMO.

Reply:
> The tests/assertions on coef and intercept, however, are veeeery essential, IMO.

Then maybe we could change the statistical structure of the dataset features to find a way to make the problem easier to identify with a limited number of samples? I am not sure how though. They are already well decorrelated.

Reply:
>Then maybe we could change the statistical structure of the dataset features to find a way to make the problem easier to identify with a limited number of samples?

I don't think so. This really is **THE** test for linear quantile regression. The data is constructed such that:
- y is asymmetric
- `X @ coef + intercept` is larger than zero
- `quantile(y, level=q) = X @ coef + intercept`, i.e. the true quantile at level q is a linear model and we know its coefficients.
- `n_features=2` is already minimal without being trivial

It could be that another distribution than the exponential converges faster. But the exponential already is very well behaved, no heavy tails like the pareto.

**Summary**
I'm out of ideas.

Reply:
Ok let's hardcode some random seed-dependent tolerance levels for the coef and intercept checks then...

Reply:
I tested for 100 random seeds and these assertions are really hard to meet with 1000 samples. As per [NEP19](https://numpy.org/neps/nep-0019-rng-policy.html), we can rely on `numpy.random.RandomState` to always produce the same random numbers with the same seed.
TLDR: I agree (unfortunately).

Reply:
Yes I had tried the same. Increasing `n_samples` makes the problem identifiable for any seed but then the test would be to slow... So we have to live with a seed dependent test for now.